app/uk/gov/hmrc/personaldetailsvalidation/PersonalDetailsValidator.scala

```diff
@@ -60,9 +60,13 @@ private class PersonalDetailsValidator[Interpretation[_] : Monad](matchingConnec
   }.leftMap { error => sendErrorEvents(personalDetails); error }
 
   private implicit class MatchResultOps(matchResult: MatchResult) {
-    def toPersonalDetailsValidation(optionallyHaving: PersonalDetails): PersonalDetailsValidation = matchResult match {
-      case MatchSuccessful(_) => PersonalDetailsValidation.successful(optionallyHaving)
-      case MatchFailed(_) => PersonalDetailsValidation.failed()
+    def toPersonalDetailsValidation(optionallyHaving: PersonalDetails): PersonalDetailsValidation = {
+      (matchResult, optionallyHaving) match {
+        case (MatchSuccessful(matchingPerson: PersonalDetailsNino), other: PersonalDetailsWithPostCode) =>
+          PersonalDetailsValidation.successful(other.addNino(matchingPerson.nino))
```

Comment:
PersonalDetailsValidation object graph doesn't need postcode. Refer this - https://github.com/hmrc/personal-details-validation/pull/13/files#r172578403